shell/src/main/java/alluxio/cli/fs/command/PersistCommand.java

```diff
@@ -47,71 +68,116 @@ public String getCommandName() {
     return "persist";
   }
 
+  @Override
+  public Options getOptions() {
+    return new Options().addOption(PARALLELISM_OPTION);
+  }
+
   @Override
   public void validateArgs(CommandLine cl) throws InvalidArgumentException {
     CommandUtils.checkNumOfArgsNoLessThan(this, cl, 1);
   }
 
   @Override
-  protected void runPlainPath(AlluxioURI plainPath, CommandLine cl)
-      throws AlluxioException, IOException {
-    persist(plainPath);
+  public String getUsage() {
+    return "persist [-p|--parallelism <#>] <path> [<path> ...]";
+  }
+
+  @Override
+  public String getDescription() {
+    return "Persists files or directories currently stored only in Alluxio to the UnderFileSystem.";
   }
 
   @Override
   public int run(CommandLine cl) throws AlluxioException, IOException {
+    int parallelism = 4;
+    if (cl.hasOption(PARALLELISM_OPTION.getLongOpt())) {
+      String parellismOption = cl.getOptionValue(PARALLELISM_OPTION.getLongOpt());
+      parallelism = Integer.parseInt(parellismOption);
+    }
     String[] args = cl.getArgs();
```

Comment:
minor: would be nice to separate argument setting and persist implementation into separate methods

Reply:
I annotated the method, splitting it didn't seem to be very useful since no one else should be calling either portion individually.